how-we-work/how-to-upgrade-dependencies.md

```diff
@@ -0,0 +1,96 @@
+# How to Upgrade Dependencies
+
+Once a month at the beginning of the month, the oncall engineer should visit the current versions of package dependencies that we have for both the frontend (`npm`) and the backend (`pip`) for `corpora-data-portal` and `cellxgene`. We do this so that we keep up to date with any security vulnerabilities that may exist in older versions and so that we don't end up too out-of-date with improvements that occur in our dependencies causing forward-incompatibility. The sections below describe the policies and instructions for each subset of our product suite.
+
+As a general rule please do not directly import the latest version of the downstream package that contains the vulnerability. For example, if we import package `A` and package `A` depends on package `B` which depends on package `C` which contains a vulnerability, please do not directly import the latest version of package `C` directly into our package. If a newer version of package `A` exists where they have addressed the vulnerability, directly upgrade `A`. Otherwise, wait until there is a new one available.
```

Comment:
This begs a policy question: if the immediately upstream dependency does not address the vuln in a timely manner, how should the vuln be resolved?  Example:  package `C` above is abandon-ware.

In some cases, the right decision will be to migrate to a different (actively supported) dependency.  In others, it will be to just wait.  And in others, we can safely ignore it. I think these likely require a case-by-case decision. 

I suggest that _when_ this occurs, the on-call engineer should file an issue, making note of the problem, and starting to track the likelihood of resolution.  At some point, if these issues are old enough, we should make that case-dependent decision.